### ISSUE TITLE
BUG: AASTex table missing \\ on last data line

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#6933]
When using the io.ascii AASTex writer, the last data line is lacking a ``\\``:

```
\startdata
a \\
b \\
cc
\enddata
```

This can result in the following LaTeX error:

```
./spw15_spectrum_gauss_fit.tex:64: Misplaced \omit.
\startdata ... {\vskip -9pt}\pt@head \par #1\omit 
                                                  \\\omit \\\hline \end {tab...
l.64 \enddata
             
? 
```

This can be fixed easily by adding ``\\`` on the end of the last data line.

cc @hamogu

**Comment by hamogu** [comment]
This is behaviour was implemented after some discussion in #3888 because ending the last line of a deluxetable with `\\` output an empty line. There are a few other issues where this was touched on again (e.g. for reading in #5427, see also #5236, #4561).

Can you post the entire table that causes the error? Also, it is possible that AASTeX actually changed the implementation of their deluxetable in the LaTeX template so it would be interesting to know which version of that you are using. 

I'm uncomfortable to revert to the pre #3888 behaviour until we understand why this problem turn up now but not before.
However, if we can't find the root cause, then I'm in for adding the final `\\` back in. It's better to produce an unwanted blank line then to occasionally have output that does not compile in LaTeX at all.

**Comment by astrofrog** [comment]
With AASTex 6.1, the following example:

```
\documentclass{aastex61}
\begin{document}

\title{}
\author{}
\date{}
\maketitle

\begin{deluxetable}{cc}
\tablehead{\colhead{a} & \colhead{b}}
\startdata
a & 1 \\
b & 2 \\
c & 3
\enddata
\end{deluxetable}

\end{document}
```

generated from:

```python
In [4]: t = Table()

In [5]: t['a'] = ['a','b','c']

In [6]: t['b'] = [1,2,3]

In [7]: t.write('test.tex', format='ascii.aastex')
```

errors, with the error shown above.

**Comment by astrofrog** [comment]
It looks like this was fixed in AASTex 6.2: https://github.com/AASJournals/AASTeX60/issues/10 - I'll close this issue since it seems that's the easiest fix.

**Comment by taldcroft** [comment]
Woohoo, here's to bugs going away with no action!
